lexibank_cosgrovevoro.py

```diff
@@ -1,81 +1,111 @@
-
-        >>> with self.raw_dir.temp_download("http://www.example.com/e.tsv", "example.tsv") as data:
-        ...     self.raw_dir.write_csv('template.csv', self.raw_dir.read_csv(data, delimiter='\t'))
-        """
+        self.raw_dir.xls2csv("ids_cl_võro.xlsx")
 
     def cmd_makecldf(self, args):
-        """
-        Convert the raw data to a CLDF dataset.
-
-        A `pylexibank.cldf.LexibankWriter` instance is available as `args.writer`. Use the methods
-        of this object to add data.
-        """
-        data = self.raw_dir.read_csv('template.csv', dicts=True)
-
-        # short cut to add concepts and languages, provided your name spaces
-        # match lexibank's expected format.
-        args.writer.add_concepts()
-        args.writer.add_languages()
-
-        # if not, then here is a more detailed way to do it:
-        #for concept in self.concepts:
-        #    args.writer.add_concept(
-        #        ID=concept['ID'],
-        #        Name=concept['ENGLISH'],
-        #        Concepticon_ID=concept['CONCEPTICON_ID']
-        #    )
-        #for language in self.languages:
-        #    args.writer.add_language(
-        #        ID=language['ID'],
-        #        Glottolog=language['Glottolog']
-        #    )
+        glottocode = "voro1241"
+        lang_id = glottocode
+        lang_name = "Võro"
+        transcription = "StandardOrth"
+        source = "cosgrove2020"
+
+        ccode = {
+            x.attributes["ids_id"]:
+                {
+                    "concepticon_id": x.concepticon_id,
+                    "concepticon_gloss": x.concepticon_gloss,
+                    "name": x.english,
+                }
+                for x in self.conceptlists[0].concepts.values()
+        }
+
+        args.writer.add_sources(*self.raw_dir.read_bib())
+
+        args.writer.add_language(
+            ID=lang_id,
+            Name=lang_name,
+            Glottocode=glottocode,
+            Contributors=['Leon Cosgrove|Author', 'Sulev Iva|Author', 'Leon Cosgrove|Data Entry'],
+            default_representation=transcription,
+            date='2020-09-17',
+        )
 
         # add data
-        for row in pylexibank.progressbar(data):
-            # .. if you have segmentable data, replace `add_form` with `add_form_with_segments`
-            # .. TODO @Mattis, when should we use add_forms_from_value() instead?
-            lex = args.writer.add_form(
-                Language_ID=row['Language_ID'],
-                Parameter_ID=row['Parameter_ID'],
-                Value=row['Word'],
-                Form=row['Word'],
-                Source=[row['Source']],
-            )
-            # add cognates -- make sure Cognateset_ID is global!
-            args.writer.add_cognate(
-                lexeme=lex,
-                Cognateset_ID=row['Cognateset_ID']
+        for form in pylexibank.progressbar(self.read_csv()):
+            args.writer.add_concept(
```

Comment:
Shouldn't `args.writer.add_concepts()` do the trick - considering that we have a related concept list in Concepticon?